src/components/WatchlistHistory/GetWatchlistHistory.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import { graphql } from 'react-apollo'
+import { connect } from 'react-redux'
+import { TOTAL_TYPES } from './constants'
+import {
+  CATEGORY_HISTORY_QUERY,
+  PROJECTS_HISTORY_QUERY,
+  WATCHLIST_HISTORY_QUERY
+} from './WatchlistHistoryGQL'
+import WatchlistHistoryWidget from './WatchlistHistoryWidget'
+import { filterEmptyStats } from './utils'
+
+const getMarketcapQuery = ({ type, projects, range, byFunction }) => {
+  const { from, to } = range.method
+
+  const slugsQueryTotal = graphql(CATEGORY_HISTORY_QUERY, {
+    props: ({ data: { historyPrice = [], loading: isLoading } }) => ({
+      historyPrice: filterEmptyStats(historyPrice),
+      isLoading
+    }),
+    options: () => ({
+      variables: {
+        from: from.toISOString(),
+        to: to.toISOString(),
+        interval: range.interval,
+        slug: TOTAL_TYPES[type]
+      }
+    })
+  })
+
+  if (TOTAL_TYPES[type]) return slugsQueryTotal
+
+  return byFunction
+    ? graphql(PROJECTS_HISTORY_QUERY, {
+      props: ({
+        data: { projectsListHistoryStats = [], loading: isLoading }
+      }) => ({
+        historyPrice: filterEmptyStats(projectsListHistoryStats),
+        isLoading
+      }),
+      options: () => ({
```

Comment:
options before props. We used it often. And you use it in the next graphql func.